lib/Runtime/Base/CompactCounters.h

```diff
@@ -129,8 +135,13 @@ namespace Js
                 }
             }
 
-            Assert(fieldSize == 4);
-            return this->fields->u32Fields[type] = val;
+            if (fieldSize == 4)
+            {
+                return this->fields->u32Fields[type] = val;
+            }
+
+            Assert(fieldSize == 0 && this->isCleaningUp && this->fields == nullptr); // OOM when allocating the counters structure
```

Comment:
yes, I thought about that. but the register default value is unsigned max (-1). let me check if we reset those counters when cleaning up. and assert val==0||val==-1 if needed